components/infra-proxy-service/server/environments.go

```diff
@@ -143,7 +142,65 @@ func (s *Server) DeleteEnvironment(ctx context.Context, req *request.Environment
 	return &response.Environment{
 		Name: environment.Name,
 	}, nil
+}
+
+// UpdateEnvironment updates the environment attributes
+func (s *Server) UpdateEnvironment(ctx context.Context, req *request.UpdateEnvironment) (*response.Environment, error) {
+	c, err := s.createClient(ctx, req.OrgId, req.ServerId)
+	if err != nil {
+		return nil, err
+	}
+
+	if req.Name == "" {
+		return nil, status.Error(codes.InvalidArgument, "must supply environment name")
+	}
+
+	cookbooks := req.CookbookVersions
+	if len(cookbooks) == 0 {
+		cookbooks = map[string]string{}
+	}
+
+	var defAtt interface{}
+	var ovrAtt interface{}
+
+	defIn := req.DefaultAttributes
+	if defIn == "" {
+		defIn = "{}"
+	}
+
+	overIn := req.OverrideAttributes
+	if overIn == "" {
+		overIn = "{}"
+	}
+	defaultAttributes := json.RawMessage(defIn)
+	overrideAttributes := json.RawMessage(overIn)
```

Comment:
This works, but have we considered using `structpb` for these? It's like the protobuf version of "arbitrary json".

Reply:
Okay, No I have not tried yet any reference would be helpful. Thanks!

Reply:
Sorry haven't got a better reference than this right now: https://stackoverflow.com/q/52966444

Reply:
What the better approach is here may depend on how large the objects are, too.

Reply:
I'll look into it, as I have used stringify JSON a couple of places so possibly going to cover in further PR. Thanks